src/Components/App/App.js

```diff
@@ -235,7 +235,7 @@ class App extends Component {
           ></Route>
           <Route
             exact 
-            path='/:activity/:results'
+            path={this.state.randomActivity ? '/:activity/:results': '/form'}
```

Comment:
I think this fixes part of the bug but we should definitely come back and do a more thorough error handling.